cli/src/main.rs

```diff
@@ -277,27 +285,26 @@ async fn maybe_find_peer(role: &ServalRole, override_var: &str) -> Result<String
         Err(_) => 8181,
     };
 
-    let metadata = PeerMetadata::new(format!("client@{host}"), vec![ServalRole::Client], None);
+    let metadata = PeerMetadata::new(
+        format!("client@{host}"),
+        "None".to_string(),
+        vec![ServalRole::Client],
+        None,
+    );
     let mut mesh = ServalMesh::new(metadata, mesh_port, None).await?;
     mesh.start().await?;
 
     // There has to be a better way.
     sleep(Duration::from_secs(20)).await;
 
-    let result = if let Some(target) = mesh.find_role(role).await {
-        if let Some(addr) = target.address() {
-            Ok(format!("http://{addr}"))
-        } else {
-            Err(anyhow!(
-                "found a peer without an address somehow: {:?}",
-                target
-            ))
-        }
+    let candidates = mesh.find_role(role).await;
+    let result = if let Some(target) = candidates.first() {
+        Ok(target.http_address().to_string())
     } else {
         Err(anyhow!("Unable to locate a peer with the {role} role"))
     };
 
-   mesh.stop().await?;
+    mesh.stop().await?;
```

Comment:
Ok, this finally made it through my skull and brings up a very interesting architectural consideration.

Nothing needs to change right now, but it's worth discussion:

Up until now, I had never contemplated the thought that an instance of the `serval` CLI tool would have to join the mesh to do any work. That totally works, of course, but it seems a bit odd for something so ephemeral to create so much work for the mesh—*everybody* is going to have to update their known peers list, and it will take things a while to settle down—all for a peer who will vanish a few seconds later.

From the perspective of the mesh, a better design would be for the `serval` CLI tool to connect to an instance of `serval-agent` running on localhost. If none is running on localhost, joining the mesh to find a node to talk to would be the fallback. Perhaps we could store all of this in `$TMP` or `/var/run` or whatever?